File: cpp/sum-of-multiples/sum_of_multiples.cpp
#include "sum_of_multiples.h"

namespace sum_of_multiples 
{
	long to(std::vector<int> factors, size_t max)
	{
		long sum = 0;
		size_t j, i;

		for (i = 1; i < max; i++)
			for (j = 0; j < factors.size(); j++)
				if (i % factors[j] == 0)
				{
					sum += i;
					break;
				}

		return sum;
	}
}  // namespace sum_of_multiples

File: java/sum-of-multiples/src/main/java/SumOfMultiples.java
import java.util.Arrays;
import java.util.stream.IntStream;

class SumOfMultiples
{
    private int sum;

    SumOfMultiples(int number, int[] set)
    {
        this.sum = calculateSumOfMultiples(number, set);
    }

    int getSum()
    {
        return sum;
    }

    private int calculateSumOfMultiples(int max, int[] factors)
    {
        return IntStream
                .range(1, max)
                .filter(i -> Arrays.stream(factors).anyMatch(factor -> factor != 0 && i % factor == 0))
                .sum();
    }
}
